Error happened when use global eslint

I use globally installed eslint and eslint-plugin-angular. My project in disk D and npm in disk C. So when I trying to use eslint-plugin-disabled I got the error below

> Error on attempt to find parent and base ESLint configuration files of '.eslintrc.json': Cannot find module 'eslint' from 'D:\***\***'
			Error: Cannot find module 'eslint' from 'D:\***\***'
			at Function.module.exports [as sync] (C:\Users\***\AppData\Roaming\npm\node_modules\eslint-plugin-disable\node_modules\resolve\lib\sync.js:40:15)
			at Object.<anonymous> (C:\Users\***\AppData\Roaming\npm\node_modules\eslint-plugin-disable\index.js:26:26)
			at Module._compile (module.js:643:30)
			at Object.Module._extensions..js (module.js:654:10)
			at Module.load (module.js:556:32)
			at tryModuleLoad (module.js:499:12)
			at Function.Module._load (module.js:491:3)
			at Module.require (module.js:587:17)
			at require (internal/module.js:11:18)
			at Plugins.load (C:\Users\***\AppData\Roaming\npm\node_modules\eslint\lib\config\plugins.js:101:26)

Config file .eslintrc.json below
```
{
  "extends":[
    "plugin:angular/johnpapa"
  ],
  "plugins": [
    "disable"
  ]
}
```

--- a/index.js
+++ b/index.js
@@ -19,29 +19,58 @@ var resolveOptions = {
   basedir: workingDirectoryPath,
 };
 
+function resolveEslint(path) {
+  // First try to locate a path on behalf of current working directory to find
+  // a local eslint installation. If it fails to find a local installation, it
+  // goes through the rest of node paths to find a package.
+  // "resolve" might not be able to find global modules on Windows,
+  // (https://github.com/mradionov/eslint-plugin-disable/issues/12),
+  // because it originally does not look in a folder with global modules.
+  // Try using Node.js "require.resolve" to find the global installation.
+  try {
+    return resolve.sync(path, resolveOptions);
+  } catch (err) {
+    // Ignore error, try using Node.js require.resolve instead.
+  }
+  try {
+    return require.resolve(path);
+  } catch (err) {
+    throw err;
+  }
+}
+
 // ESLint does not pass config object to processors.
 // Locate and use ESLint engine to be able to load ESLint config for particular
 // files.
-// Find ESLint installation used in current working directory
-var eslintPath = resolve.sync('eslint', resolveOptions);
-var eslintOptionsPath = resolve.sync('eslint/lib/options', resolveOptions);
 
-var eslint = require(eslintPath);
-var eslintOptions = require(eslintOptionsPath);
+var eslint = null;
+var eslintOptions = null;
 
-var cliArgs = process.argv;
+try {
+  var eslintPath = resolveEslint('eslint', resolveOptions);
+  var eslintOptionsPath = resolveEslint('eslint/lib/options', resolveOptions);
+
+  eslint = require(eslintPath);
+  eslintOptions = require(eslintOptionsPath);
+} catch (err) {
+  console.error('[eslint-plugin-disable]', err);
+}
 
 var engine = null;
 
-// Extra guard if engine fails to load. Must catch it here, because
-// otherwise ESLint will consider that plugin itself throws an error and
-// fails to load.
-// Options also might throw an error in case some of them are incorrect.
-try {
-  var options = eslintOptions.parse(cliArgs);
-  engine = new eslint.CLIEngine(translateOptions(options));
-} catch (err) {
-  console.error('[eslint-plugin-disable]', err);
+if (eslint && eslintOptions) {
+
+  // Extra guard if engine fails to load. Must catch it here, because
+  // otherwise ESLint will consider that plugin itself throws an error and
+  // fails to load.
+  // Options also might throw an error in case some of them are incorrect.
+  try {
+    var cliArgs = process.argv;
+    var options = eslintOptions.parse(cliArgs);
+    engine = new eslint.CLIEngine(translateOptions(options));
+  } catch (err) {
+    console.error('[eslint-plugin-disable]', err);
+  }
 }
 
 var processors = {};

